stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/ExchangeDeclare.java

```diff
@@ -55,7 +55,7 @@ public static Object exchangeDeclare(Strand strand, ObjectValue channelObjectVal
         try {
             ChannelUtils.exchangeDeclare(channel, exchangeConfig);
             if (transactionContext != null) {
-                transactionContext.handleTransactionBlock();
+                transactionContext.handleTransactionBlock(strand);
```

Comment:
Why are we always checking whether a transaction context is present before checking whether we are inside a transaction?
I think it would be better if we first, check whether we are inside a transaction block and then check whether there is a transaction context.